Src/FluentAssertions/Formatting/StringValueFormatter.cs

```diff
@@ -18,13 +18,6 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
     {
         string result = "\"" + value + "\"";
 
-        if (context.UseLineBreaks)
-        {
-            formattedGraph.AddFragmentOnNewLine(result);
-        }
-        else
-        {
-            formattedGraph.AddFragment(result);
-        }
+        formattedGraph.AddFragment(result);
```

Comment:
🏕️ I think we should try to format this as:

```
Expected subject to refer to 
{ 
    UserName = "JohnDoe" 
} 
because they are the same, but found 
{ 
     Name = "John Doe" 
}.
```

Notice the extra new-line after the closing brace. It would be nice if this PR could fix that as well.

Reply:
This seems tricky as the message template comes from lots of different constraints and I don't think there's a consistent place to put the newline.

For example, `Should().Be()`:
https://github.com/fluentassertions/fluentassertions/blob/649e9fd0054627927427959e618a9a56f9cee251/Src/FluentAssertions/Numeric/ComparableTypeAssertions.cs#L57

I don't think we'd want to put the newline in the formatting of then object itself, because then it would come at the end of the `{0}` and `{1}` substitutions.  That would be OK for the `{0}` object when there was a `because` string, but would leave a newline before a comma otherwise - and would always leave a newline just before the full stop of the `{1}` output.

We would have to put the newline in all the message templates, wouldn't we?  And it may be dependant on how the values ended up being formatted as empty anonymous types are all on one line anyway.

Reply:
Hmm, you may be right.